tools/python_tools/cuvslam_tools/dataset_preparation/rgbd.py

```
@@ -0,0 +1,501 @@
+        try:
+            values = [float(column) for column in columns[1:]]
+        except ValueError as exc:
+            raise RgbdConversionError(
+                f"{source} line {line_number}: invalid numeric value"
+            ) from exc
+        # float() accepts "inf" and "nan". Left alone they propagate into every
+        # interpolated pose as silently wrong ground truth rather than an error.
+        if not all(math.isfinite(value) for value in values):
+            raise RgbdConversionError(
+                f"{source} line {line_number}: non-finite pose value"
+            )
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)